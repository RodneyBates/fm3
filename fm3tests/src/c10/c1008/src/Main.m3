
(* -----------------------------------------------------------------------1- *)
(* This file is part of the FM3 Modula-3 compiler.                           *)
(* Copyright 2024        Rodney M. Bates.                                    *)
(* rodney.m.bates@acm.org                                                    *)
(* Licensed under the MIT License.                                           *)
(* -----------------------------------------------------------------------2- *)

(* Test variable declarations, some with CT errors. *) 

MODULE Main

; TYPE T = LONG

; VAR V0 : T
; VAR INTEGER : T  (* Reserved id declared. *)
; VAR V : REAL 
; VAR V0 : CHAR    (* Duplicate declaration . *) 
; VAR U : BOOLEAN

; VAR
    V1 : T 
  ; INTEGER : T (* Reserved id declared. *)
  ; V2 : REAL 
  ; V1 : CHAR    (* Duplicate declaration . *) 
  ; U1 : BOOLEAN

  ; V3 : BOOLEAN := TRUE
  ; V4 := FALSE
  ; V5            (* Neither type nor initial value. *) 
  

; BEGIN
  END Main 
.
