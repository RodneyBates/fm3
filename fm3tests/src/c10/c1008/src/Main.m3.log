[31mERROR: [0mMain.m3 (16,6) Identifier "INTEGER" is predefined and cannot be redeclared (2.8.2).
[31mERROR: [0mMain.m3 (23,4) Identifier "INTEGER" is predefined and cannot be redeclared (2.8.2).
[31mERROR: [0mMain.m3[30,4] Variable must have a type and/or an initial value. (2.4.3)
[31mERROR: [0mMain.m3 (25,4) Duplicate declaration of "V1", ignored, original at (22,4).
[31mERROR: [0mMain.m3 (18,6) Duplicate declaration of "V0", ignored, original at (15,6).
