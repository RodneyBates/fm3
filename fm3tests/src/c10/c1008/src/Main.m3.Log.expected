[31mERROR: [0mMain.m3[13,7] Duplicate declaration of "T", previously introduced at [11,0],
           original declaration at Main.i3:[13,7], (2.5.1)
[31mERROR: [0mMain.m3[16,6] Reserved identifier "INTEGER" cannot be declared (2.8.2).
[31mERROR: [0mMain.m3[23,4] Reserved identifier "INTEGER" cannot be declared (2.8.2).
[31mERROR: [0mMain.m3[30,4] Variable must have a type and/or an initial value (2.4.3)
